blscurve/bls_backend.nim

```diff
@@ -27,13 +27,20 @@ when BLS_FORCE_BACKEND == "blst" or (
       defined(amd64) and
       gorgeEx(getEnv("CC", "gcc") & " -march=native -dM -E -x c /dev/null | grep -q SSSE3").exitCode == 0))
   ):
+  # BLST supports: x86_64 and ARM64
+  # and has optimized SHA256 routines for x86_64 CPU with SSE3
   const BLS_BACKEND* = BLST
+elif BLS_FORCE_BACKEND == "auto" and defined(amd64):
```

Comment:
You're missing `BLS_FORCE_BACKEND == "blst"` here. And doesn't `__BLST_PORTABLE__` also work on ARM and on 32-bit arches?

Reply:
The branch before is always taken when we force `"blst"` (whether GCC returns true or false).

`__BLST_PORTABLE__` is only used on x86_64 because not all x86_64 supports SSE3 while all ARM64 supports the required Neon vectorized instructions.
![image](https://user-images.githubusercontent.com/22738317/93525819-d9738c00-f936-11ea-8580-b18d05ce042e.png)